src/utils/client.ts

```diff
@@ -373,9 +374,18 @@ export default class Client {
       method: 'PATCH',
       body: JSON.stringify(data)
     };
-    return this._fetch(url, options).then((response) => {
-      return response ? response.json() : Promise.resolve();
-    });
+    return this._fetch(url, options)
+      .then((response) => {
+        return response ? response.json() : Promise.resolve();
+      })
+      .then((data: any) => {
+        if (data === undefined) return Promise.resolve();
+        Object.entries<boolean>(data.form_completed).forEach(
+          ([formKey, formCompleted]) =>
+            authState.setFormCompleted(formKey, formCompleted)
```

Comment:
instead of needing a custom authState.setFormCompleted function, could we update the session form_copmleted flag directly then call rerenderAllForms?